test/tokengenerator_test.go

```diff
@@ -0,0 +1,147 @@
+package testutils
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/dgrijalva/jwt-go"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
+	jose "gopkg.in/square/go-jose.v2"
+)
+
+func TestTokenGeneratorKeys(t *testing.T) {
+
+	t.Run("create keys", func(t *testing.T) {
+		tokengenerator := NewTokenGenerator()
+		kid0 := uuid.NewV4().String()
+		key0, err := tokengenerator.CreateKey(kid0)
+		require.NoError(t, err)
+		kid1 := uuid.NewV4().String()
+		key1, err := tokengenerator.CreateKey(kid1)
+		require.NoError(t, err)
+		// check key equality by comparing the modulus
+		require.NotEqual(t, key0.N, key1.N)
+	})
+
+	t.Run("get key", func(t *testing.T) {
+		tokengenerator := NewTokenGenerator()
+		kid0 := uuid.NewV4().String()
+		key0, err := tokengenerator.CreateKey(kid0)
+		require.NoError(t, err)
```

Comment:
you should also verify that key0Retrieved and key1Retrieved are not nil

Reply:
fixed here: 3af5601b2df6e621740160430d5a76981b66115c